support-workers/src/main/scala/com/gu/config/Configuration.scala

```diff
@@ -19,9 +22,16 @@ object Configuration {
 
   SafeLogger.info(s"Load from S3: $loadFromS3, Stage: $stage")
 
-  val config = PrivateConfigLoader
-    .forEnvironment(loadFromS3)
-    .load(stage, ConfigFactory.load())
+  // this is static so it persists between lambda executions, but lazy so it doesn't cause a fatal error at class loading time
+  private lazy val loadConfig = PrivateConfigLoader
+    .forEnvironment(Configuration.loadFromS3)
+    .load(Configuration.stage, ConfigFactory.load())
```

Comment:
configfactory.load uses the current thread context classloader.  However this is not recommended when running code within SBT (test or run) as in a Future pool `the context classloader of the thread that backs the future is the scala library classloader which is not able to load project classes`
https://www.scala-sbt.org/1.x/docs/In-Process-Classloaders.html#Troubleshooting

So I need to work out what that really means, it seems bizarre that typesafe config and sbt tests wouldn't work in a Future, so there must be a proper solution.

Reply:
😱

Reply:
I have fixed it to specifiy the classloader used to load the current class.  I discovered someone else hit the same issue with sbt 1.3 (sbt does fancy classloader stuff for perf reasons)
https://www.jan0sch.de/post/typesafe-config-sbt-classloader-issues/

Reply:
the current thread classloader seems to be designed so that parent application servers can load things from the child classloader, so for example play loading application config.  That may explain why typesafe config defaults to loading from the thread loader instead of the current one.
More reading if anyone is interested:
https://www.infoworld.com/article/2077344/find-a-way-out-of-the-classloader-maze.html